sync2jira/upstream_issue.py

```diff
@@ -358,6 +383,26 @@ def github_issues(upstream, config):
         if issue.get('milestone', None):
             issue['milestone'] = issue['milestone']['title']
 
+        if not issue.get('storypoints', None):
+            issue['storypoints'] = ''
+            orgname, reponame = upstream.rsplit('/', 1)
+            issuenumber = issue['number']
+            default_github_project_fields = config['sync2jira']['default_github_project_fields']
+            project_github_project_fields = config['sync2jira']['map']['github'][upstream]['github_project_fields']
+            github_project_fields = default_github_project_fields | project_github_project_fields
+            variables = {"orgname": orgname, "reponame": reponame, "issuenumber": issuenumber}
+            for fieldname, values in github_project_fields.items():
+                ghfieldname, _ = values
+                variables['ghfieldname'] = ghfieldname
+                response = requests.post(graphqlurl, headers=headers, json={"query": project_items_query, "variables": variables})
+                data = response.json()
+                if fieldname == 'storypoints':
```

Comment:
Since this block only cares about `'storypoints'` (based on the `if` tests at lines 386 and 399) I think the loop at line 394 is unnecessary.  I believe you can replace the line with
```python
           fieldname, values = 'storypoints', github_project_fields['storypoints']
```
(or just assign `values` and use `'storypoints'` in place of `fieldname` in the loop body code).

On the other hand, if you want this code to be generalized to handle other fields in addition to `'storypoints'`, then you need to remove the conditional at line 399 and replace lines 386-394 with something like
```python
        github_project_fields = default_github_project_fields | github_config[upstream]['github_project_fields']
        if any(issue.get(f) for f in github_project_fields):
            orgname, reponame = upstream.rsplit('/', 1)
            variables = {"orgname": orgname, "reponame": reponame, "issuenumber": issue['number']}
            for fieldname, values in github_project_fields.items():
```
after hoisting the assignment of `default_github_project_fields` out of the loop at line 333 (since it doesn't change) and adding this definition next to it:
```python
        github_config = config['sync2jira']['map']['github']
```

And, you might want to replace the loop at line 394 with something like this
```python
            for fieldname, values in ((f, v) for f, v in github_project_fields.items() if issue.get(f)):
```
to avoid fields which aren't in `issue`.

(And, if you are going to generalize the code here, you probably want to do something similar at lines 292-293.)

Reply:
However, one thing to note or think about:  my code suggestion above probably won't handle the case where the upstream issue is updated to change a field from non-blank to blank/empty -- my code will not explicitly propagate that change to the downstream...I don't know whether that leaves the downstream unchanged or whether it will be implicitly cleared.

Reply:
OK; hopefully this will all fall out when you add 'priority'.